### DISCUSSION TITLE
Installing multiple versions of an application (.NET SDK)

### DISCUSSION THREAD

**Discussion by dnperfors** [microsoft/winget-cli#755] [Q&A]
I recently started to use winget on a new machine and so far I like it, however I am wondering how to actually install multiple versions of an application.
In my case I would like to have installed both .NET Core SDK 3.1 and .NET SDK 5, however I can't seem to find a way using winget to do this although it would be very helpful for keeping the sdk up to date.

**Comment by jedieaston** [comment]
They should probably be separated into separate packages, I've been thinking about this since right now if you install OpenJDK you get version 15 while many/most people probably want version 11 (as that's the LTS version). Of course, you can do --version but then once ` winget upgrade `  works it'll be bothering you about it until the end of time.

**Comment by denelon** [comment]
The PIN feature https://github.com/microsoft/winget-cli/issues/476 should help reduce the problems associated with a developer wanting to retain a specific version of a package. We will have to make sure PIN is respected across upgrade and import / export as well.
